pkg/internal/checkup/executor/executor.go

```diff
@@ -41,307 +35,26 @@ type vmiSerialConsoleClient interface {
 	VMISerialConsole(namespace, name string, timeout time.Duration) (kubecli.StreamInterface, error)
 }
 
-type podExecuteClient interface {
-	ExecuteCommandOnPod(ctx context.Context, namespace, name, containerName string, command []string) (stdout, stderr string, err error)
-}
-
-type testPmdPortStats struct {
-	RXPackets int64
-	RXDropped int64
-	RXTotal   int64
-	TXPackets int64
-	TXDropped int64
-	TXTotal   int64
-}
-
-type testPmdStatsIdx int
-
-const (
-	testPmdStatsPort0 testPmdStatsIdx = iota
-	testPmdStatsPort1
-	testPmdPortStatsSummary
-	testPmdPortStatsSize
-)
-
 type Executor struct {
-	client                           vmiSerialConsoleClient
-	podClient                        podExecuteClient
-	namespace                        string
-	vmiUsername                      string
-	vmiPassword                      string
-	vmiEastNICPCIAddress             string
-	vmiEastEthPeerMACAddress         string
-	vmiWestNICPCIAddress             string
-	vmiWestEthPeerMACAddress         string
-	testDuration                     time.Duration
-	verbosePrintsEnabled             bool
-	trafficGeneratorPacketsPerSecond string
+	client      vmiSerialConsoleClient
+	namespace   string
+	vmiUsername string
+	vmiPassword string
 }
 
-const testpmdPrompt = "testpmd> "
-
-func New(client vmiSerialConsoleClient, podClient podExecuteClient, namespace string, cfg config.Config) Executor {
+func New(client vmiSerialConsoleClient, namespace string) Executor {
 	return Executor{
-		client:                           client,
-		podClient:                        podClient,
-		namespace:                        namespace,
-		vmiUsername:                      config.VMIUsername,
-		vmiPassword:                      config.VMIPassword,
-		vmiEastNICPCIAddress:             config.VMIEastNICPCIAddress,
-		vmiEastEthPeerMACAddress:         cfg.TrafficGeneratorEastMacAddress.String(),
-		vmiWestNICPCIAddress:             config.VMIWestNICPCIAddress,
-		vmiWestEthPeerMACAddress:         cfg.TrafficGeneratorWestMacAddress.String(),
-		testDuration:                     cfg.TestDuration,
-		verbosePrintsEnabled:             cfg.Verbose,
-		trafficGeneratorPacketsPerSecond: cfg.TrafficGeneratorPacketsPerSecond,
+		client:      client,
+		namespace:   namespace,
+		vmiUsername: config.VMIUsername,
```

Comment:
why do you keep LoginToCentOS? We will need to revise that as well in future commits.

Reply:
It is the minimal usage of the serial console, I don't think it needs to be removed (but will probably be changed).
I don't see how we can not login to the guests.